lib/src/repo_path.rs

```diff
@@ -51,6 +72,45 @@ impl From<String> for RepoPathComponentBuf {
     }
 }
 
+impl AsRef<RepoPathComponent> for RepoPathComponent {
+    fn as_ref(&self) -> &RepoPathComponent {
+        self
+    }
```

Comment:
added missing `AsRef<T> for T` (where `T = RepoPathComponent`)
https://doc.rust-lang.org/std/convert/trait.AsRef.html#reflexivity